tests/dashboard/test_dashboard.py

```diff
@@ -127,3 +127,73 @@ async def should_allow_access(
     assert body["users"][0]["user"]["firstName"] == "User2"
     assert body["users"][1]["user"]["lastName"] == "Foo"
     assert body["users"][1]["user"]["firstName"] == "User1"
+
+
+async def test_connection_uri_has_http_prefix_if_localhost(app: TestClient):
+    connection_uri = start_st()
+    connection_uri_without_protocol = connection_uri.replace("http://", "")
+
+    st_args = get_st_init_args(
+        [
+            session.init(get_token_transfer_method=lambda _, __, ___: "cookie"),
+            dashboard.init(api_key="someKey"),
+        ]
+    )
+
+    st_config = st_args.get("supertokens_config")
+    if st_config:
+        st_config.connection_uri = connection_uri_without_protocol
+
+    init(**st_args)
+
+    res = app.get(url="/auth/dashboard")
+    assert res.status_code == 200
+    assert f'window.connectionURI = "{connection_uri}"' in str(res.text)
+
+
+async def test_connection_uri_has_https_prefix_if_not_localhost(app: TestClient):
+    start_st()
+    connection_uri = "https://try.supertokens.com"
+    connection_uri_without_protocol = connection_uri.replace("https://", "")
+
+    st_args = get_st_init_args(
+        [
+            session.init(get_token_transfer_method=lambda _, __, ___: "cookie"),
+            dashboard.init(api_key="someKey"),
+        ]
+    )
+
+    st_config = st_args.get("supertokens_config")
+    if st_config:
+        st_config.connection_uri = connection_uri_without_protocol
+
+    init(**st_args)
+
+    res = app.get(url="/auth/dashboard")
+    assert res.status_code == 200
+    assert f'window.connectionURI = "{connection_uri}"' in str(res.text)
+
+
+async def test_that_first_connection_uri_is_selected_among_multiple_uris(
+    app: TestClient,
+):
+    first_connection_uri = start_st()
+    second_connection_uri = "https://try.supertokens.com"
+    multiple_connection_uris = f"{first_connection_uri};{second_connection_uri}"
+
+    st_args = get_st_init_args(
+        [
+            session.init(get_token_transfer_method=lambda _, __, ___: "cookie"),
+            dashboard.init(api_key="someKey"),
+        ]
+    )
+
+    st_config = st_args.get("supertokens_config")
+    if st_config:
+        st_config.connection_uri = multiple_connection_uris
+
+    init(**st_args)
```

Comment:
does this get init with multiple_connection_uris? since you change connection_uri in st_config which is a copy of st_args?

Reply:
Yeah it does. I put a log in the `implementation.py` and you can see the logs in the attached screenshots.

<img width="1241" alt="image" src="https://github.com/supertokens/supertokens-python/assets/22813027/884b35ed-6646-4444-8a9e-3c95a51ba203">


<img width="1435" alt="image" src="https://github.com/supertokens/supertokens-python/assets/22813027/94622cfd-13f8-4130-b5ea-de96ea17dadb">